.github/workflows/sel4test-hw.yml

```diff
@@ -0,0 +1,124 @@
+# Copyright 2021, Proofcraft Pty Ltd
+#
+# SPDX-License-Identifier: BSD-2-Clause
+
+# sel4test hardware builds and runs
+#
+# See sel4test-hw/builds.yml in the repo seL4/ci-actions for configs.
+
+name: seL4Test-HW
+
+on:
+  workflow_call:
+
+# intended to run on
+#  pull_request_target:
+#    types: [opened, reopened, synchronize, labeled]
+# needs PR target for secrets access; guard by requiring label
+
+# downgrade permissions to read-only as you would have in a standard PR action
+permissions:
+  contents: read
+
+jobs:
+  code:
+    name: Freeze Code
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'pull_request_target' &&
+              github.event.action != 'labeled' &&
+              (contains(github.event.pull_request.labels.*.name, 'hw-build') ||
+               contains(github.event.pull_request.labels.*.name, 'hw-test')) ||
+            github.event_name == 'pull_request_target' &&
```

Comment:
Isn't this redundant?

Reply:
if the workflow is used correctly probably yes, but if we use a wrong trigger then the rest of the event context is different and not safe to access, so I though it better to make sure it does not trigger at all instead of triggering based on data that might or might not make sense.

Reply:
I meat grouping the labeld/not-labeld closer together using this:
```
github.event_name == 'pull_request_target' &&
((github.event.action != 'labeled' &&
  (contains(github.event.pull_request.labels.*.name, 'hw-build') ||
   contains(github.event.pull_request.labels.*.name, 'hw-test'))) ||
 (github.event.action == 'labeled' &&
  (github.event.label.name == 'hw-build' ||
   github.event.label.name == 'hw-test')))

```

Reply:
Yes, that would be fine. Maybe slightly easier to read.